src/components/protected_route.tsx

```diff
@@ -3,24 +3,28 @@
 import { useRouter } from "next/navigation";
 import { useAuthState } from "react-firebase-hooks/auth";
 import { auth } from "@/logic/fb";
+import { use, useEffect } from "react";
 
-export default function ProtectedRoute({ children }: { children: React.ReactNode }) {
+export default function ProtectedRoute({
+  children,
+}: {
+  children: React.ReactNode;
+}) {
   const [user, loading, error] = useAuthState(auth);
   const router = useRouter();
 
-  if (user) {
-    return <>{children}</>;
-  }
+  useEffect(() => {
+    router.prefetch("/login");
```

Comment:
Myślę że to powinno mieć własny useEffect bo jest całkowite niezwiązane z resztą tego effectu i niepotrzebnie wykonuje się wiele razy.